src/tls.rs

```diff
@@ -284,6 +285,10 @@ impl Transport for TlsStream {
     fn remote_addr(&self) -> Option<SocketAddr> {
         Some(self.remote_addr)
     }
+
+    fn peer_certificates(&self) -> PeerCertificates {
+        self.peer_certs.clone()
```

Comment:
`clone` is generally better to avoid. Return a reference and let the caller decide if it wants to copy it.

Reply:
This is from the original PR. `PeerCertificates` is actually an `Arc` pointer, so I think the `clone` isn't too bad, right? And, not surprisingly, when I remove the `clone`, I get all sorts of nasty errors :-)